pkg/collector/python/init.go

```diff
@@ -372,9 +376,16 @@ func Initialize(paths ...string) error {
 
 // Destroy destroys the loaded Python interpreter initialized by 'Initialize'
 func Destroy() {
-	if rtloader != nil {
-		C.destroy(rtloader)
+	// Sanity check - this should ideally never happen
+	if rtloader == nil {
+		log.Warn("Python runtime already destroyed. Ignoring action.")
+		return
 	}
+
+	log.Info("Destroying Python runtime")
+	C.destroy(rtloader)
```

Comment:
Can `Destroy` be called by 2 goroutines? If yes, there is a thread safe issue and `C.destroy(rtloader)` may called twice.